content/post/078-go-ipfs-0.4.23.md

```diff
@@ -0,0 +1,84 @@
+
+### 😱 Fixed Panics and Crashes
+
+Panicing won't help, in life, and also in golang. Stay calm and breathe slowly. We patched a number of panics and crashes that were uncovered, including a panic seen commonly in the websocket transport. High ten 🙌?
+
+### 🔁 Fixed Resursive Resolving of dnsaddr Multiaddrs
+
+Imagine this: rolling out a brand spanking new set of bootstrap nodes only to discover their new addresses are not resolvable. I know right - bad news bears 🐻!?
+
+Ok, "bear" this in mind: you want to keep all your DNS TXT records [below 512 bytes to avoid UDP fragmentation](https://serverfault.com/questions/840241/do-dns-queries-always-travel-over-udp), otherwise you'll get a truncated reply and have to connect with TCP to get all the records. If you have lots of dnsaddr TXT records then it can be more efficient to use recursive resolving than to get a truncated reply and go through the famous 18-way SYN, SYN-ACK ACK, ACK-SYN, ACK-ACK (...etc, etc) TCP handshake, not to mention the fact that go-ipfs will not even try to fallback to TCP 😅.
+
+Anyway, long story short. We fixed that. You're welcome.
+
+### 📻 Retuned Connection Manager
+
+The Connection Manager has been tuned to better prioritise existing connections by not counting new connections in the "grace" period towards connection limits. New connections are like new friends. You can't hang out with everyone all the time, I mean, it just gets difficult to book a resturant after a while.
+
+You also wouldn't stop being friends with Jane just because you met Sarah _once_ on the train. You and Jane have history, think of everything you've been through. Remember that time when Jane's dog, Dave, ran away? I know, it's a weird name for a dog, I mean who gives a human name to a dog anyway, but I guess that's one of the reasons you like Jane. Anyway, she lost her dog and you both looked all around town for it, you were about to give up but then you heared faint wimpering as you were walking back to the house. Dave had somehow managed to fall into the old abandoned well!
+
+You see?! History! ...and, erh, what was I saying? Oh yeah, Connection Manager - new connections don't cause us to close useful, existing connections (like Jane), but it does mean you'll keep more connections in total. Maybe consider reducing the `HighWater` setting in your config.
+
+### 🍖 Reduced Relay Related DHT Spam
+
+When `AutoRelay` was enabled, your IPFS node basically spent all it's time searching the DHT for relays, like a fly to a honey trap. I say that because every other IPFS node was doing the same and if you had `AutoRelay` and `RelayHop` enabled then, erm, DoS happens. Bad times.
```

Comment:
We should try to explain why.

Previously, when AutoRelay was enabled, an unreachable (e.g., behind a NAT) go-ipfs would try to find 3 nodes announcing themselves as relays in the DHT. Once found, go-ipfs would connect to these nodes and advertise them as relays.

Unfortunately, many of these public nodes were poorly configured (e.g., with very low connection limits) and unreliable (i.e., they'd constantly kill connections to peers). So, instead of finding a set of relays and sticking with them, go-ipfs would _continuously_ search for better relays causing 95% of the DHT traffic.